src/components/App/App.js

```diff
@@ -14,34 +15,33 @@ import ProjectPage from '../ProjectPage';
 import CreateNewProjectWizard from '../CreateNewProjectWizard';
 import ProjectConfigurationModal from '../ProjectConfigurationModal';
 import Initialization from '../Initialization';
+import LoadingScreen from '../LoadingScreen';
 
 import type { Project } from '../../types';
 
 type Props = {
   selectedProjectId: ?Project,
+  showLoadingScreen: boolean,
 };
 
 class App extends PureComponent<Props> {
   render() {
-    const { selectedProjectId } = this.props;
-
+    const { selectedProjectId, showLoadingScreen } = this.props;
+    console.log(showLoadingScreen);
     return (
       <Initialization>
         {wasSuccessfullyInitialized =>
           wasSuccessfullyInitialized && (
             <Fragment>
               <Titlebar />
-
               <Wrapper>
                 <ApplicationMenu />
-
+                {showLoadingScreen && <LoadingScreen />}
```

Comment:
I would connect to Redux inside `LoadingScreen` component, so it'easier to read here. Similar to how it is done by the `ProjectConfigurationModal`.